### PR TITLE
Misc addons 12.0 odoo backup sh

### PR DESCRIPTION
**Pull request by GabbasovDinar**

### CODE REVIEW

File: odoo_backup_sh/controllers/main.py [hunk 1]
```diff
@@ -93,10 +93,17 @@ def update_cloud_params(cls, cloud_params, redirect, call_from):
         if not user_key:
             user_key = ''.join(random.choice(string.hexdigits) for _ in range(30))
             cls.set_config_values('options', {'odoo_backup_user_key': user_key})
-        cloud_params = requests.get(BACKUP_SERVICE_ENDPOINT + '/get_cloud_params', params={
-            'user_key': user_key,
-            'redirect': redirect,
-        }).json()
+
+        # If your website does not have a certificate and you are not sending sensitive data,
+        # you can set the verify parameter to False and requests will ignore SSL verification.
```
Comment (yelizariev) [inline odoo_backup_sh/controllers/main.py]: It actually sends sensetive data

File: odoo_backup_sh/controllers/main.py [hunk 2]
```diff
@@ -93,10 +93,17 @@ def update_cloud_params(cls, cloud_params, redirect, call_from):
...
+        try:
+            cloud_params = requests.get(BACKUP_SERVICE_ENDPOINT + '/get_cloud_params', params={
+                'user_key': user_key,
+                'redirect': redirect,
+            }, verify=False).json()
+        except Exception:
+            _logger.exception('Failed to load cloud params')
```
Comment (yelizariev) [inline odoo_backup_sh/controllers/main.py]: This try block doesn't prevent error raising later (cloud_params is undefined

File: odoo_backup_sh/controllers/main.py [hunk 3]
```diff
@@ -78,7 +78,7 @@ def get_cloud_params(cls, redirect=None, call_from='backend'):
         cloud_params = cls.get_config_values('options', [
             'odoo_backup_user_key', 'amazon_bucket_name', 'amazon_access_key_id', 'amazon_secret_access_key',
             'odoo_oauth_uid'])
-        if None in cloud_params.values():
+        if None or '' in cloud_params.values():
```
Comment (yelizariev) [inline odoo_backup_sh/controllers/main.py]: Don't send such updates to the branch. You can make it locally if needed

File: odoo_backup_sh/models/odoo_backup_sh.py [hunk 4]
```diff
@@ -35,6 +35,7 @@
 class BackupConfig(models.Model):
     _name = 'odoo_backup_sh.config'
     _description = 'Backup Configurations'
+    _rec_name = 'database'
```
Comment (yelizariev) [inline odoo_backup_sh/models/odoo_backup_sh.py]: :+1:

File: odoo_backup_sh/models/odoo_backup_sh.py [hunk 5]
```diff
@@ -183,15 +184,21 @@ def update_info(self, cloud_params):
...
-                                {'Key': '%s/%s' % (cloud_params['odoo_oauth_uid'], file_name)} for file_name in
-                                remote_backups[backup_config.database][backup_dt]
-                            ]
-                            del remote_backups[backup_config.database][backup_dt]
+                    # TODO: check it (I fixed the error below)
+                    # File "/mnt/addons/it-projects-llc/misc-addons/odoo_backup_sh/models/odoo_backup_sh.py", line 186, in update_info
+                    # backup_dts = copy.deepcopy(remote_backups[backup_config.database])
+                    # KeyError: 'backup'
+                    remote_db = remote_backups.get(backup_config.database, False)
+                    if remote_db:
```
Comment (yelizariev) [inline odoo_backup_sh/models/odoo_backup_sh.py]: Avoid indent changing. You can do instead
```

if not remote_db:
    continue
```

File: odoo_backup_sh/controllers/main.py (hunk 3 above)
Comment (yelizariev) [reply odoo_backup_sh/controllers/main.py]:
```
In [2]: not all([x for x in [1,2,3]])
Out[2]: False

In [3]: not all([x for x in [1,2,0]])
Out[3]: True

In [4]: not all([x for x in [1,2,'']])
Out[4]: True

In [5]: not all([x for x in [1,2,False]])
Out[5]: True

```

File: odoo_backup_sh/controllers/main.py (hunk 2 above)
Comment (GabbasovDinar) [reply odoo_backup_sh/controllers/main.py]: @yelizariev yes, but now we can open the Backups menu. Without this construction we will get exeption error therefore the Backups menu will not open

**Pull request merged**